ZEngine/include/ZEngine/Managers/ShaderManager.h

```diff
@@ -27,5 +27,15 @@ namespace ZEngine::Managers {
          * @return A shader instance
          */
         Ref<Rendering::Shaders::Shader> Load(const char* filename) override;
+
+        static const std::string GetFragmentFilename(const std::string_view key);
+        static const std::string GetVertexFilename(const std::string_view key);
+
+        static Ref<Rendering::Shaders::Shader> Get(ZEngine::Rendering::Specifications::ShaderSpecification spec);
```

Comment:
```suggestion
        static Ref<Rendering::Shaders::Shader> Get(const ZEngine::Rendering::Specifications::ShaderSpecification& spec);
```

Reply:
shaderspecification cannot be a const variable because we will be updating the spec.verterxfilename and fragment file name in the Get function

Reply:
oh okay, mark it ref then